clang/include/clang/Basic/BuiltinsNVPTX.def

```diff
@@ -650,6 +663,1378 @@ BUILTIN(__nvvm_atom_cas_gen_ll, "LLiLLiD*LLiLLi", "n")
 TARGET_BUILTIN(__nvvm_atom_cta_cas_gen_ll, "LLiLLiD*LLiLLi", "n", SM_60)
 TARGET_BUILTIN(__nvvm_atom_sys_cas_gen_ll, "LLiLLiD*LLiLLi", "n", SM_60)
 
+TARGET_BUILTIN(__nvvm_atom_acquire_add_gen_i, "iiD*i", "n", SM_70)
+TARGET_BUILTIN(__nvvm_atom_cta_acquire_add_gen_i, "iiD*i", "n", SM_70)
```

Comment:
In the PTX instruction the scope appears before the semantic. I am okay with the space appearing at the end, but I think it would make more sense to have the other other options occur in the same order as the instruction.

Reply:
> In the PTX instruction the scope appears before the semantic.

Do you mean the other way around? In builtins in this PR the scope appears before the semantic and in PTX ISA the semantic appears before the scope.

I would not change the order of anything but the semantics, as existing builtins have other parts in that order. There are no builtins for atomics with semantic, so I guess we can put that in any place.

Reply:
> Do you mean the other way around? In builtins in this PR the scope appears before the semantic and in PTX ISA the semantic appears before the scope.

Sorry, that is what I meant yes!

> I would not change the order of anything but the semantics, as existing builtins have other parts in that order. There are no builtins for atomics with semantic, so I guess we can put that in any place.

I agree, if other builtins use a specific order I think it should be kept as consistent as possible, but as you mention there are no other other builtins with atomic semantics (at least that I know off) so personally I would prefer to have them as similar to the corresponding PTX instruction as possible to make the mapping simpler.

Reply:
I refactored this as requested.

Reply:
Thanks!